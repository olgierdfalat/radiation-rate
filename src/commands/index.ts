export * from './checksum';
export * from './filterFiles';
export * from './wrongFiles';
export * from './excelExport';
