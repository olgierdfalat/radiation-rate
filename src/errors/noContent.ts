export class NoContent extends Error {
  constructor(message?: string) {
      super(message || 'No content'); // 'Error' breaks prototype chain here
      Object.setPrototypeOf(this, new.target.prototype); // restore prototype chain
  }
}