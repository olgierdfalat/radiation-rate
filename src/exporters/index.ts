export * from './worksheetsDataProvider';
export * from './excel';