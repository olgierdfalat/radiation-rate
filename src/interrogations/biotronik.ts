import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class Biotronik extends Interrogation {
  async getData(): Promise<parsers.BiotronikData> {
    const content = await this.getContent();
    return new parsers.BiotronikData(content, this.filePath);
  }
}