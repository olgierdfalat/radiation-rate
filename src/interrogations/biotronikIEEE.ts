import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class BiotronikIEEE extends Interrogation {
  async getData(): Promise<parsers.BiotronikIEEEData> {
    const content = await this.getContent();
    return new parsers.BiotronikIEEEData(content, this.filePath);
  }
}