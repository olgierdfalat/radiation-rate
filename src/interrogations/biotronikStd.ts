import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class BiotronikStd extends Interrogation {
  async getData(): Promise<parsers.BiotronikStdData> {
    const content = await this.getContent();
    return new parsers.BiotronikStdData(content, this.filePath);
  }
}