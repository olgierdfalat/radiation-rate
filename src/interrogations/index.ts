export * from './interrogation';
export * from './stJude';
export * from './biotronikStd';
export * from '../exporters/worksheetsDataProvider';
export * from './interrogationFactory';