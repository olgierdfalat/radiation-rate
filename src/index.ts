import { Command } from 'commander';
import * as commands from './commands';


const command = new Command();
command
  .version('0.0.1')
  .description('CLI for converting heart valves output data into Excel format')
  .option('-c, --checksum <device>', 'generates checksums of all log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.checksum)
  .option('-e, --excluded <device>', 'generates list of excluded log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.excludedFiles)
  .option('-i, --included <device>', 'generates list of included log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.includedFiles)
  .option('-w, --invalid <device>', 'generates list of wrong log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.invalidFiles)
  .parse(process.argv);

