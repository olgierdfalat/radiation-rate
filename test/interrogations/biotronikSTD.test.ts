import * as path from 'path';
import * as interrogations from '../../src/interrogations';
import * as errors from '../../src/errors';


describe('interrogation Biotronik STD class', () => {
  it('should get structured data row from file', async () => {
    const filePath = path.join(__dirname, '../../../radiation-rate-data/test/Biotronik/BIOSTD_2016-12-23_12-12-56_ANONYMOUS_A_60485754.xml');
    const biotronikStd = new interrogations.BiotronikStd(filePath);
    const data = await biotronikStd.getData();
    expect(data.getRow()).toMatchSnapshot();
  });

  it('should throw No Content exception for empty log file', async () => {
    const filePath = path.join(__dirname, '../fixtures/empty-invalid.log');
    const biotronikStd = new interrogations.BiotronikStd(filePath);
    await expect(biotronikStd.getData()).rejects.toThrow(errors.NoContent);
  });
});