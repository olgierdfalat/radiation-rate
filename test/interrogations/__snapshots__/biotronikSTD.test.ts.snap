// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interrogation Biotronik STD class should get structured data row from file 1`] = `
Object {
  "admissionDate": Object {
    "name": "AdmissionDate",
    "type": "date",
    "value": "2016-12-23",
  },
  "admissionTime": Object {
    "name": "AdmissionTime",
    "type": "time",
    "value": "12:12:00",
  },
  "examinationDate": Object {
    "name": "ExaminationDate",
    "type": "date",
    "value": "2016-12-23",
  },
  "examinationFunctionalDomain": Object {
    "name": "FunctionalDomain",
    "type": "string",
    "value": "DEFI",
  },
  "examinationTime": Object {
    "name": "ExaminationTime",
    "type": "time",
    "value": "12:12:00",
  },
  "interfaceDataCreationDate": Object {
    "name": "CreationDate",
    "type": "date",
    "value": "2016-12-23",
  },
  "interfaceDataCreationTime": Object {
    "name": "CreationTime",
    "type": "time",
    "value": "12:15:30",
  },
  "interfaceDataDestination": Object {
    "name": "Destination",
    "type": "string",
    "value": "Centricity Xi2",
  },
  "interfaceDataSource": Object {
    "name": "Source",
    "type": "string",
    "value": "BIOTRONIK SE & Co. KG",
  },
  "personalDataDateOfBirth": Object {
    "name": "DateOfBirth",
    "type": "date",
    "value": "1949-07-10",
  },
  "personalDataFirstName": Object {
    "name": "FirstName",
    "type": "string",
    "value": "ANONYMOUS",
  },
  "personalDataLastName": Object {
    "name": "LastName",
    "type": "string",
    "value": "ANONYMOUS",
  },
}
`;
